internal/controller/controllers/bmh_agent_controller.go

```diff
@@ -755,6 +775,16 @@ func (r *BMACReconciler) reconcileSpokeBMH(ctx context.Context, log logrus.Field
 		Namespace: agent.Spec.ClusterDeploymentName.Namespace,
 		Name:      fmt.Sprintf(adminKubeConfigStringTemplate, cd.Name),
 	}
+	isNonePlatform, err := r.isNonePlatform(ctx, cd)
+	if err != nil {
+		log.WithError(err).Errorf("Failed to determine if platform is none for cluster deployment %s/%s", cd.Namespace, cd.Name)
+	}
+	if isNonePlatform || err != nil {
+		// If none platform do not attempt to create spoke BMH
+		// If error occurred no retry will be attempted since we will get same result
+		return reconcileComplete{}
```

Comment:
We chatted about this so I want to bring here what we discussed: 

* This logic should ignore not found errors and mark the reconcile loop as complete
* The rest of the errors can be propagated (return reconcileError{})
* When errors are returned, the reconcile [loop will retry again](https://cluster-api.sigs.k8s.io/developer/providers/implementers-guide/controllers_and_reconciliation.html#reconciliation)